### PR TITLE
Update documentation - refactor and add digital twin feature

### PR DESCRIPTION
**Pull request by bikamani**
Add digital twin for features and update readme

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -49,12 +49,12 @@ Devices and data sources in an IoT solution can range from a simple network-conn
...
+| [Authentication](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-security-deployment)                     | :heavy_check_mark:       | :heavy_check_mark:*      | :heavy_check_mark:      | :heavy_check_mark:*      | :heavy_check_mark:*       | Connect your device to IoT Hub securely with supported authentication, including private key, SASToken, X-509 Self Signed over MQTT, AMQPS and HTTPS, and X-509 Certificate Authority (CA) Signed. *IoT Hub only supports X-509 CA Signed over AMQP and MQTT at the moment. X509-CA authentication over websocket and HTTPS are not supported.                                                     |
```
Comment (timtay-microsoft) [inline readme.md:52]: Java SDK supports CA signed certs over MQTT_WS and HTTPS. Just not over AMQPS_WS

File: readme.md [hunk 2]
```diff
@@ -49,12 +49,12 @@ Devices and data sources in an IoT solution can range from a simple network-conn
...
+| [Send device-to-cloud message](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-messages-d2c)     | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:*      | :heavy_check_mark:      | Send device-to-cloud messages (max 256KB) to IoT Hub with the option to add custom properties.  *IoT Hub only supports batch send over HTTPS only at the moment. The MQTT and AMQP implementation loops over the batch and sends each message individually.                           |
```
Comment (timtay-microsoft) [inline readme.md:53]: IoT Hub supports batch messages over AMQP. It's just that this SDK doesn't. You should clarify that

File: readme.md [hunk 3]
```diff
@@ -49,12 +49,12 @@ Devices and data sources in an IoT solution can range from a simple network-conn
...
+| [Direct Methods](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods)                 | :heavy_check_mark:      | :heavy_check_mark:      | :heavy_check_mark:      | :heavy_check_mark:      | :heavy_minus_sign:       | IoT Hub gives you the ability to invoke direct methods on devices from the cloud.  The SDK supports handler for generic operation.                                                                                                                                                                                                          |
+| [Upload file to Blob](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-file-upload)               | :heavy_minus_sign:       | :heavy_minus_sign:       | :heavy_minus_sign:       | :heavy_minus_sign:       | :heavy_check_mark:*      | A device can initiate a file upload and notifies IoT Hub when the upload is complete.   File upload requires HTTPS connection, but can be initiated from client using any protocol for other operations such as telemetry.  *Upload file to Blob with X509 certificate is currently not supported.                                                                        |
```
Comment (timtay-microsoft) [inline readme.md:57]: File upload can be done with x509 auth. Self signed and CA signed

**Review by timtay-microsoft — APPROVED** [review]
Just a few small comments, but otherwise looks good

File: readme.md (hunk 1 above)
Comment (bikamani) [reply readme.md:52]: I'll update it

File: readme.md (hunk 2 above)
Comment (bikamani) [reply readme.md:53]: That's a good point

**Pull request merged**